src/v/cluster/controller_backend.cc

```diff
@@ -1344,13 +1258,16 @@ ss::future<std::error_code> controller_backend::create_partition(
     // if topic is read replica, the value from remote topic manifest is
     // used
     auto initial_rev = _topics.local().get_initial_revision(ntp);
-    if (!initial_rev) {
+    auto replica_view = _topics.local().get_replicas_view(ntp);
+    if (!replica_view || !initial_rev) {
```

Comment:
Why is the replicas_view check needed? Presumably this check has been done by the caller (reconcile_ntp_step)